GiEnJul/ClientApp/src/components/InputFields/Validators/InputValidator.tsx

```diff
@@ -1,193 +1,194 @@
 ﻿import { TextField } from "@material-ui/core";
 import * as React from "react";
+import { useCallback } from "react";
 import { FC, useEffect, useState } from "react";
 import SelectInput from "../SelectInput";
 
 interface InputValidatorProps {
```

Comment:
I agree that the InputValidator adds some complexity to our solution, but I think the pros of using it overcome the cons.
Some pros is listed below:
- Custom error messages for each validation check. E.g. one for invalid email and another for no email.
- The isValid variables. These make it possible to make more flexible validation and feedback. In the institution form these is for e.g. the dinner->annet field.
- The material ui TextField doesn't support pattern. Also the "tel" type doesn't support Norwegian telephone numbers (in the way we write them).

There is a way to use material ui with validation though by using the react-material-ui-form-validator package. The problem with this package is that you can't add custom validations. There is also no support for solutions like the one in dinner-annet on the institution page. Because of these facts I think the InputValidator is more suitable.

Reply:
Maybe @Antonio-Cabrero got some input on this.

After our talk we do see that InputValidator handles much of the operations that native components could do, but it provides some added flexibility.